Gems/Atom/Utils/Code/Source/ImageComparison.cpp

```diff
@@ -21,7 +21,8 @@ namespace AZ
             const int16_t diffR = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[index]) - aznumeric_cast<int16_t>(bufferB[index])));
             const int16_t diffG = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[index + 1]) - aznumeric_cast<int16_t>(bufferB[index + 1])));
             const int16_t diffB = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[index + 2]) - aznumeric_cast<int16_t>(bufferB[index + 2])));
-            return AZ::GetMax(AZ::GetMax(diffR, diffG), diffB);
+            const int16_t diffA = static_cast<int16_t>(abs(aznumeric_cast<int16_t>(bufferA[index + 3]) - aznumeric_cast<int16_t>(bufferB[index + 3])));
```

Comment:
Do we need to check the size of the span here in case the buffers are RGB images instead of RGBA?

Reply:
This function is only used in contexts that assume a pixel size of 4 bytes: 

https://github.com/o3de/o3de/blob/2855c9197521363ab3b731ea549ee04f6b70b149/Gems/Atom/Utils/Code/Source/ImageComparison.cpp#L43

https://github.com/o3de/o3de-atom-sampleviewer/blob/d06eceefa35cd8f8b84490107e43563e020207b0/Gem/Code/Source/Automation/ScriptReporter.cpp#L1208